app/resources/v1/users.js

```diff
@@ -242,8 +248,39 @@ exports.get = async function (req, res) {
     }
   }
 
-  // TODO this function seems like it could be replaced by
-  // sequelize findByPK
+  const BTPTokenCheck = async function (userData) {
+    // check for coil provider to set access token to stream payments
+    const coilData = userData.identities.find(
+      (item) => item.provider === 'coil'
+    )
+    if (coilData !== undefined) {
+      // fetch and return btpToken
+      let btpToken = await getBTPToken(coilData.access_token)
+      if (typeof btpToken === 'undefined') {
+        // token may be expired, so lets refresh and try
+        const newAccessToken = await refreshAccessToken(coilData.refresh_token)
+
+        const identities = userData.identities
+        coilData.access_token = newAccessToken
+        addOrUpdateByProviderName(identities, coilData)
```

Comment:
I'm not a fan of functions that modify its input variables. As I have a tendency to want to replace this with table code anyway, it's not super important to fix, but for reference, I would much prefer to see something like this:

```js
const updatedIdentities = addOrUpdateByProviderName(identities, coilData) // The original `identities` is not modified

await User.update({
  identities: updatedIdentities
}, { where: { ... }})
```